### PR TITLE
[PY] chore: LangChain Sample and bugfixes

### PR DESCRIPTION
**Pull request by aacebo**
## Linked issues

#minor

## Details

- fix minor Teams AI bugs
- add update_activity adapter override as stopgap bug fix
- add langchain planner sample

### CODE REVIEW

File: python/samples/07.planners.a.langchain/infra/azure.parameters.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      "value": "${{BOT_ID}}"
+    },
+    "botAadAppClientSecret": {
+      "value": "${{SECRET_BOT_PASSWORD}}"
+    },
+    "webAppSKU": {
+      "value": "B1"
+    },
+    "botDisplayName": {
+      "value": "EchoBotInfra"
```
Comment (corinagum) [inline python/samples/07.planners.a.langchain/infra/azure.parameters.json:18]: Update name

File: python/packages/ai/teams/teams_adapter.py [hunk 2]
```diff
@@ -109,6 +109,34 @@ async def process(
 
         return res
 
+    # [TODO] remove once version 4.15.1 is installed
```
Comment (lilyydu) [inline python/packages/ai/teams/teams_adapter.py:112]: Do you mind making a ticket to track this so we don't forget?

File: python/samples/07.planners.a.langchain/infra/azure.parameters.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      "value": "${{BOT_ID}}"
+    },
+    "botAadAppClientSecret": {
+      "value": "${{SECRET_BOT_PASSWORD}}"
+    },
+    "webAppSKU": {
+      "value": "B1"
+    },
+    "botDisplayName": {
+      "value": "LightBotInfra"
```
Comment (lilyydu) [inline python/samples/07.planners.a.langchain/infra/azure.parameters.json:18]: Update to `LangChainInfra`

File: python/samples/07.planners.a.langchain/src/bot.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    time.sleep(time_ms / 1000)
+    return "done pausing"
+
+
+@app.ai.action("LightStatus")
+async def on_lights_status(
+    _context: ActionTurnContext[Dict[str, Any]],
+    state: AppTurnState,
+):
+    return "the lights are on" if state.conversation.lights_on else "the lights are off"
```
Comment (lilyydu) [inline python/samples/07.planners.a.langchain/src/bot.py:124]: do one of the 2 (prompt functions vs actions) take precedence when getting the light status?

Comment (aacebo) [reply python/samples/07.planners.a.langchain/src/bot.py:124]: they are called at different times and for different purposes so no

**Review by corinagum — APPROVED** [review]

**Review by lilyydu — APPROVED** [review]

**Pull request merged**